text/0085-api-keys.md

```diff
@@ -0,0 +1,547 @@
+        "documents.delete" //Has now access to documents deletion
+    ],
+    "indexes": [
+        "products",
+        "reviews"
+    ], //Has now access to reviews
+    "expiresAt": "2021-12-31T23:59:59Z" //Extended to the end of the year
+}
+```
+
+##### Response
+
+`200 Success`
+
+```json
+{
+    "description": "Manage Products/Reviews Documents API key",
+    "key": "d0552b41536279a0ad88bd595327b96f01176a60c2243e906c52ac02375f9bc4",
+    "actions": [
+        "documents.add",
+        "documents.delete"
+    ],
+    "indexes": [
+        "products",
+        "reviews"
+    ],
+    "expiresAt": "2021-12-31T23:59:59Z",
+    "createdAt": "2021-11-12T10:00:00Z",
+    "updatedAt": "2021-10-12T15:00:00Z"
+}
+```
+
+##### Errors
+
+- 🔴 Accessing this route without the `Authorization` header returns a [missing_authorization_header](0061-error-format-and-definitions.md#missing_authorization_header) error.
+- 🔴 Accessing this route with a key that does not have permissions (i.e. other than the master-key) returns an [invalid_api_key](0061-error-format-and-definitions.md#invalid_api_key) error.
+- 🔴 Attempting to access an API key that does not exist returns a [api_key_not_found](0061-error-format-and-definitions.md#api_key_not_found) error.
+- 🔴 Omitting Content-Type header returns a [missing_content_type](0061-error-format-and-definitions.md#missing_content_type) error.
+- 🔴 Sending an empty Content-Type returns an [invalid_content_type](0061-error-format-and-definitions.md#invalid_content_type) error.
+- 🔴 Sending a different Content-Type than `application/json` returns an [invalid_content_type](0061-error-format-and-definitions.md#invalid_content_type) error.
+- 🔴 Sending an empty payload returns a [missing_payload](0061-error-format-and-definitions.md#missing_payload) error.
+- 🔴 Sending a different payload type than the Content-Type header returns a [malformed_payload](0061-error-format-and-definitions.md#malformed_payload) error.
+- 🔴 Sending an invalid json format returns a [malformed_payload](0061-error-format-and-definitions.md#malformed_payload) error.
+- 🔴 Sending an invalid value for the `actions` field returns an [invalid_api_key_actions](0061-error-format-and-definitions.md#invalid_api_key_actions) error.
+- 🔴 Sending an invalid value for the `indexes` field returns an [invalid_api_key_indexes](0061-error-format-and-definitions.md#invalid_api_key_indexes) error.
+- 🔴 Sending an invalid value for the `expiresAt` field returns an [invalid_api_key_expires_at](0061-error-format-and-definitions.md#invalid_api_key_expires_at) error.
+- 🔴 Sending an invalid value for the `description` field returns an [invalid_api_key_description](0061-error-format-and-definitions.md#invalid_api_key_description) error.
+
+---
+
+#### **As `Anna 👩`, I want to list the API Keys**
+
+##### Request Definition
+
+`GET` - `/keys`
+
+##### Headers
+
+```
+"Authorization: Bearer :masterKey"
+```
+
+##### Response
+
+`200 Success`
+
+```json
+{
+    "results": [
```

Comment:
I'm not against the idea at all, just for curiosity: Did you add it in a `results` key in order to be able to add new fields at the root in the future?

Reply:
This is a convention that we will have from now on when the API lists resources. It actually allows us to add fields in the future quite easily. This is also what we did on the `task` list.

Reply:
Very good!